gensim/utils.py

```diff
@@ -1212,8 +1216,8 @@ def run(self):
             if self.as_numpy:
                 # HACK XXX convert documents to numpy arrays, to save memory.
                 # This also gives a scipy warning at runtime:
-                # "UserWarning: indices array has non-integer dtype (float64)"
-                wrapped_chunk = [[np.asarray(doc) for doc in chunk]]
+                # "UserWarning: indices array has non-integer dtype (float32)"
+                wrapped_chunk = [[np.asarray(doc, dtype=self.dtype) for doc in chunk]]
```

Comment:
This changes the existing logic from float64 to float32, right?

It's probably OK, but we have to make sure to mention it in the release notes, because this could lead to nasty hidden incompatibilities and bugs for existing users, who rely on double precision.